### PR TITLE
feat(plugin): datadog host/port referenceable

### PR DESCRIPTION
**Pull request by StarlightIbuki**
### Checklist

- [x] The Pull Request has tests
- [x] There's an entry in the CHANGELOG
- [x] There is a user-facing docs PR https://github.com/Kong/docs.konghq.com/pull/5353

### Full changelog

* Allow referenceable fields to be parsed as string/number/boolean
* Make Datadog host/port referenceable

### Issue reference

Fix FTI-4910

### CODE REVIEW

File: kong/plugins/datadog/schema.lua
Comment (fffonion) [inline kong/plugins/datadog/schema.lua:77]: let's not do the number change for now, the default value being null issue happens only on string field.

**Comment by StarlightIbuki** [comment]
As the client requested, we are taking over this.

File: spec/03-plugins/08-datadog/01-log_spec.lua
Comment (fffonion) [inline spec/03-plugins/08-datadog/01-log_spec.lua:217]: I feel we can use a different env var, so that if we accidently run into the code path that "use the env var when `host` is unset", it will help us catch the regression

**Comment by hbagdi** [comment]
This requires documentation, especially for kubernetes users since this change significantly increases the usability of the plugin on k8s. Please create a docs PR and block this PR until that PR is merged.

**Comment by hbagdi** [comment]
@fffonion Please review.

**Review by hanshuebner — APPROVED** [review]

**Comment by hbagdi** [comment]
@StarlightIbuki Please cherry-pick this to EE.
And one question: are we planning to make the same change to opentelemetry plugin?

**Pull request merged**
